src/Generator/Program.fs

```diff
@@ -259,12 +271,44 @@ let main argv =
     let typecheckedFunctions = looslyTypedFunctionsToTypedFunctions enumMap functions
     printfn "overall correct function specifications: %d" typecheckedFunctions.Length
     let openGlSpecVersions = getEnumCasesAndCommandsPerVersion test
-    for currOpenGL_Spec in openGlSpecVersions do
-        generateEnums enums
-        generateInterface typecheckedFunctions
-        generateStaticClass typecheckedFunctions
-        generateDummyTypes()
+    let basePath = "./"
+    let inline writeToFile openGlVersion topic content =
+        let pathToFile = basePath </> openGlVersion
+        let fileName = topic + ".cs"
+        Directory.CreateDirectory pathToFile |> ignore
+        File.WriteAllText(pathToFile </> fileName, content)
+    openGlSpecVersions 
+    |> Array.Parallel.iter (fun currOpenGL_Spec ->
```

Comment:
just a hint, when the name of a collection is `items` i expect the item to be named `item`. so in this case, rather than having `currOpenGL_Spec` `glSpec` might be a better name for the item in an enumeration called `glSpecs` rather than `openGlSpecVersions`.